rust/tool/src/generation.rs

```diff
@@ -40,10 +40,19 @@ pub struct Generation {
 impl Generation {
     pub fn from_link(link: &GenerationLink) -> Result<Self> {
         let bootspec_path = link.path.join("boot.json");
-        let generation: BootspecGeneration = serde_json::from_slice(
-            &fs::read(bootspec_path).context("Failed to read bootspec file")?,
-        )
-        .context("Failed to parse bootspec json")?;
+        let generation = fs::read(bootspec_path)
+            .with_context(|| "Failed to read bootspec")
```

Comment:
Nit: you probably don't need the closure here since you're not actually computing anything; usually it's fine to use `context("static string")`, but you  want to reach for `with_context` when you want to do something like `format!("yaddah yaddah {something}, {}", some_expensive_computation_that_should_only_be_done_once())`

Doesn't really matter either way, but figured I'd offer my thoughts :)

Reply:
I thought `with_context` would "push" the context into the stack trace of contexts, but I was wrong, you're right :).